### PR TITLE
 sql: fix flaky size error in max_size session serialization test

### LINKED ISSUE #85821: sql: flake on TestSessionMigration/max_size session serialization size error
In a PR with a new session variable I sometimes get this CI error:

```
 TestSessionMigration/max_size
    session_migration_test.go:103:
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/testdata/session_migration/max_size:13: SELECT encode(crdb_internal.serialize_session(), 'hex')
        expected:
        ERROR: crdb_internal.serialize_session(): serialized session size 5.0 KiB exceeds max allowed size 2.0 KiB (SQLSTATE 54000)
        found:
        ERROR: crdb_internal.serialize_session(): serialized session size 5.1 KiB exceeds max allowed size 2.0 KiB (SQLSTATE 54000)
    --- FAIL: TestSessionMigration/max_size (0.41s)
```

Jira issue: CRDB-18460

### PR DESCRIPTION
**Pull request by cucaroach**
Fixes: #85821
    
Release note: None

### CODE REVIEW

File: pkg/sql/session_migration_test.go [hunk 1]
```diff
@@ -209,6 +210,30 @@ WHERE dump.variable IS NULL OR dump2.variable IS NULL OR dump.variable != dump2.
...
+						errorRE = arg.Vals[0]
+					}
+				}
+				if errorRE == "" {
+					t.Fatalf("error requires regexp arg")
+				}
+				_, err := dbConn.Exec(ctx, getQuery())
+				if err != nil {
+					if ok, err := regexp.MatchString(errorRE, err.Error()); ok {
+						return "error matched successfully"
```
Comment (rafiss) [inline pkg/sql/session_migration_test.go:230]: nit: i think it'd be simpler to return an empty string here, but not a blocking comment.

**Review by rafiss — APPROVED** [review]
thanks! left an optional nit

File: pkg/sql/session_migration_test.go (hunk 1 above)
Comment (cucaroach) [reply pkg/sql/session_migration_test.go:230]: :+1:

**Pull request merged**